test/engine/enginerpc/enginerpc.cc

```diff
@@ -153,6 +153,28 @@ TEST_F(EngineRpc, GetVersion) {
   erpc.shutdown();
 }
 
+TEST_F(EngineRpc, GetHost) {
+  enginerpc erpc("0.0.0.0", 40001);
+  std::atomic<bool> continuerunning(true);
+
+  auto fn = [&continuerunning]() {
+    while (continuerunning) {
+      command_manager::instance().execute();
+      std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    }
+  };
+
+  std::thread th(fn);
```

Comment:
I know, we already validated a code like this one.

I think it would be a good exercise to use a condition_variable here. You would not need this sleep_for call and you would not need continuerunning to be atomic.

with condition_variables, you can give directly the condition while waitintg... Try it!!